daml-lf/spec/daml-lf-1.rst

```diff
@@ -1955,187 +1975,326 @@ exact output.
     —————————————————————————————————————————————————————————————————————— EvExpFromAnySucc
-      'from_any' @τ e ‖ E₀  ⇓  'Some' @τ v ‖ E₁
+      'from_any' @τ e  ⇓  Ok ('Some' @τ v)
 
-      e ‖ E₀  ⇓  Ok ('to_any' @τ₁ v) ‖ E₁     τ₁ ≠ τ₂
+      e  ⇓  Ok ('to_any' @τ₁ v)     τ₁ ≠ τ₂
     —————————————————————————————————————————————————————————————————————— EvExpFromAnyFail
-      'from_any' @τ₂ e ‖ E₀  ⇓  'None' ‖ E₁
+      'from_any' @τ₂ e  ⇓  Ok 'None'
+
+      e₁  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpCaseErr
+      'case' e₁ 'of' {  p₁ → e₁ | … |  pₙ → eₙ }  ⇓  Err t
 
-      e₁ ‖ E₀  ⇓  Ok v₁ ‖ E₁
+      e₁  ⇓  Ok v₁
       v 'matches' p₁  ⇝  Succ (x₁ ↦ v₁ · … · xₘ ↦ vₘ · ε)
-      e₁[x₁ ↦ v₁, …, xₘ ↦ vₘ] ‖ E₁  ⇓  r ‖ E₂
+      e₁[x₁ ↦ v₁, …, xₘ ↦ vₘ]  ⇓  r
     —————————————————————————————————————————————————————————————————————— EvExpCaseSucc
-      'case' e₁ 'of' {  p₁ → e₁ | … |  pₙ → eₙ } ‖ E₀  ⇓  r ‖ E₂
+      'case' e₁ 'of' {  p₁ → e₁ | … |  pₙ → eₙ }  ⇓  r
 
-      e₁ ‖ E₀  ⇓  Ok v₁ ‖ E₁    v₁ 'matches' p₁  ⇝  Fail
-      'case' v₁ 'of' { p₂ → e₂ … | pₙ → eₙ } ‖ E₁  ⇓  r ‖ E₂
+      e₁  ⇓  Ok v₁    v₁ 'matches' p₁  ⇝  Fail
+      'case' v₁ 'of' { p₂ → e₂ … | pₙ → eₙ }  ⇓  r
     —————————————————————————————————————————————————————————————————————— EvExpCaseFail
-      'case' e₁ 'of' { p₁ → e₁ | p₂ → e₂ | … | pₙ → eₙ } ‖ E₀
-        ⇓
-      r ‖ E₂
+      'case' e₁ 'of' { p₁ → e₁ | p₂ → e₂ | … | pₙ → eₙ } ⇓ r
 
-      e₁ ‖ E₀  ⇓  Ok v₁ ‖ E₁     v 'matches' p  ⇝  Fail
-    —————————————————————————————————————————————————————————————————————— EvExpCaseErr
-      'case' e₁ 'of' { p → e } ‖ E₀  ⇓  Err "match error" ‖ E₁
+      e₁  ⇓  Ok v₁     v 'matches' p  ⇝  Fail
+    —————————————————————————————————————————————————————————————————————— EvExpCaseEmpty
+      'case' e₁ 'of' { p → e }  ⇓  Err "match error"
+
+       eₕ  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpConsErr1
+      'Cons' @τ eₕ eₜ  ⇓  Err t
 
-       eₕ ‖ E₀  ⇓  Ok vₕ ‖ E₁
-       eₜ ‖ E₁  ⇓  Ok vₜ ‖ E₂
+       eₕ  ⇓  Ok vₕ
+       eₜ  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpConsErr2
+      'Cons' @τ eₕ eₜ  ⇓  Err t
+
+       eₕ  ⇓  Ok vₕ
+       eₜ  ⇓  Ok vₜ
     —————————————————————————————————————————————————————————————————————— EvExpCons
-      'Cons' @τ eₕ eₜ ‖ E₀  ⇓  Ok ('Cons' @τ vₕ vₜ) ‖ E₂
+      'Cons' @τ eₕ eₜ  ⇓  Ok ('Cons' @τ vₕ vₜ)
+
+       e  ⇓  Err t
+    —————————————————————————————————————————————————————————————————————— EvExpSomeErr
+      'Some' @τ e  ⇓  Err t
 
-       e ‖ E₀  ⇓  Ok v ‖ E₁
+       e  ⇓  Ok v
     —————————————————————————————————————————————————————————————————————— EvExpSome
-      'Some' @τ e ‖ E₀  ⇓  Ok ('Some' @τ v) ‖ E₂
+      'Some' @τ e  ⇓  Ok ('Some' @τ v)
+
+      𝕋(F) = ∀ (α₁: ⋆). … ∀ (αₘ: ⋆). σ₁ → … → σₙ → σ
+      e₁  ⇓  Ok v₁
+        ⋮
+      eᵢ  ⇓  Err t
```

Comment:
Should we make it clearer that `e_2, ..., e_{i-1}` need to succeed as well? The same occurs a few more times below.

Reply:
Would it be clear enough as:

```
      e₁  ⇓  Ok v₁
        ⋮
      eᵢ₋₁  ⇓  Ok ᵢ₋₁
      eᵢ  ⇓  Err t
```

Reply:
Yes, that should make it crystal clear. (There's a `v` missing in the penultimate line.)